src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -102,8 +98,12 @@ public function __construct(
         $realEncoders = [];
         foreach ($encoders as $encoder) {
             if ($encoder instanceof SerializerAwareInterface) {
+                trigger_deprecation('symfony/serializer', '7.1', 'Interface %s is deprecated,  use %s instead.', SerializerAwareInterface::class, NormalizerAwareInterface::class);
```

Comment:
What about moving that `trigger_deprecation` to the `SerializerAwareInterface` instead?

Reply:
Yeah.. and we cannot really do that in a good way while keep things working. 

I could stop using the trait but add `setSerializer()` on the classes that uses the trait.